### PR TITLE
Conditional age disclaimer

### PR DESCRIPTION
**Pull request by mxsnyder**
Addresses #305 

- Adds a disclaimer to the list, details, and registration views that children under 8 require an adult.

### CODE REVIEW

File: js/apps/events-page/constants.js [hunk 1]
```diff
@@ -27,7 +27,11 @@
 			eventDetailsPartial: '/_js/apps/events-page/partials/eventDetails.html',
 			eventRegistrationPartial: '/_js/apps/events-page/partials/eventRegistration.html'
 		},
-		requestChunkSize: 10
+		requestChunkSize: 10,
+		messages: {
```
Comment (martypowell) [inline js/apps/events-page/constants.js]: i know this is kinda a pain, but **should** this maybe be reoganzied a bit

```javascript
ageDisclaimer: {
  message: '...',
  groups: [...]
}
```

File: js/apps/events-page/directives/eventsList.js [hunk 2]
```diff
@@ -16,6 +16,8 @@
 				dateUtilityService.formatSchedule(eventItem, eventItem.EventLength, eventItem.AllDay);
 
 			innerScope.getDisplayDate = (eventGroup) => eventGroup.date.toLocaleDateString('en-US', dateSettings);
+
+			innerScope.shouldShowDisclaimer = eventItem => ageDisclaimerService.shouldShowDisclaimer(eventItem);
```
Comment (martypowell) [inline js/apps/events-page/directives/eventsList.js]: can this just be?

```
innerScope.shouldShowDisclaimer =  ageDisclaimerService.shouldShowDisclaimer;
```

File: js/apps/events-page/directives/templates/eventsList.html [hunk 3]
```diff
@@ -43,6 +43,13 @@
 								</div>						
 							</div>
 						</div>
+						<div class="row" ng-if="shouldShowDisclaimer(eventItem)">
```
Comment (martypowell) [inline js/apps/events-page/directives/templates/eventsList.html]: this doesn't have a controller alias, is that right?

File: js/apps/events-page/services/ageDisclaimerService.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+((app) => {
+	const ageDisclaimerService = ($window, CONSTANTS) => {
+		const shouldShowDisclaimer = (eventItem) => {
+			const ageGroupsForDisclaimer = CONSTANTS.ageGroupsForDisclaimer;
+			const ageGroupsFromEvent = eventItem.AgeGroups;
+
+			const intersection = $window._.intersection(ageGroupsForDisclaimer, ageGroupsFromEvent);
+
+			return intersection > 0;
```
Comment (martypowell) [inline js/apps/events-page/services/ageDisclaimerService.js]: this is really cool, I really need to get into lodash

**Comment by mxsnyder** [comment]
I hereby declare this to be ready for another look.

File: js/apps/events-page/directives/eventsList.js (hunk 2 above)
Comment (mxsnyder) [reply js/apps/events-page/directives/eventsList.js]: Done

File: js/apps/events-page/constants.js (hunk 1 above)
Comment (mxsnyder) [reply js/apps/events-page/constants.js]: Done!

**Pull request merged**